### ISSUE TITLE
several cases with unnecessary temporary objects not detected by `modernize-use-emplace`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#79213]
```cpp
#include <map>
#include <utility>

void f()
{
    std::map<int, int> m;
    m.insert(std::make_pair(0, 0));
    m.insert(std::pair<int, int>(0, 0));
    m.emplace(std::pair<int, int>(0, 0));

    m.emplace(std::make_pair(0, 0));
}
```

```
<source>:11:15: warning: unnecessary temporary object created while calling emplace [modernize-use-emplace]
   11 |     m.emplace(std::make_pair(0, 0));
      |               ^~~~~~~~~~~~~~~    ~
```

https://godbolt.org/z/8MhYPv9qP
